packages/pool/contracts/StakeUtils.sol

```diff
@@ -81,43 +77,6 @@ contract StakeUtils is TransferUtils, IStakeUtils {
             userStakedNow >= amount,
             ERROR_VALUE
             );
-
-        // Revoke the reward of the current epoch if applicable
-        uint256 currentEpoch = block.timestamp / EPOCH_LENGTH;
-        if (!user.epochIndexToRewardRevocationStatus[currentEpoch])
-        {
-            Reward storage currentReward = epochIndexToReward[currentEpoch];
-            if (currentReward.amount != 0)
-            {
-                uint256 tokensToRevoke = currentReward.amount
-                     * getValueAt(user.shares, currentReward.atBlock)
-                     / getValueAt(totalShares, currentReward.atBlock);
-                uint256 sharesToBurn = (tokensToRevoke * totalSharesNow
-                    + (userSharesNow * totalStake)
-                    - (userStakedNow * totalSharesNow))
-                    / (totalStake + tokensToRevoke - userStakedNow);
-                if (sharesToBurn != 0)
-                {
-                    // Do not allow the user to burn what they are trying to
-                    // unstake
-                    require(sharesToBurn < userSharesNow, ERROR_UNAUTHORIZED);
-                    // The reward gets redistributed to the current stakers
-                    // Note that the lock for this reward will remain
-                    userSharesNow = userSharesNow - sharesToBurn;
-                    totalSharesNow = totalSharesNow - sharesToBurn;
-                    user.shares.push(Checkpoint({
-                        fromBlock: block.number,
-                        value: userSharesNow
-                        }));
-                    totalShares.push(Checkpoint({
-                        fromBlock: block.number,
-                        value: totalSharesNow
-                        }));
```

Comment:
Not only does this penalty cost a lot of gas, it also bloats `user.shares` and `totalShares` even more